flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/utils/FactoryUtils.java

```diff
@@ -0,0 +1,80 @@
+package com.datastrato.gravitino.flink.connector.utils;
+
+import static org.apache.flink.table.factories.FactoryUtil.validateFactoryOptions;
+import static org.apache.flink.table.factories.FactoryUtil.validateWatermarkOptions;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.flink.table.factories.CatalogFactory;
+import org.apache.flink.table.factories.Factory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FactoryUtils {
+
+  private FactoryUtils() {}
+
+  private static final Logger LOG = LoggerFactory.getLogger(FactoryUtils.class);
+
+  /** Utility for working with {@link Factory}s. */
+  public static class GravitinoCatalogFactoryHelper extends FactoryUtil.CatalogFactoryHelper {
+
+    private GravitinoCatalogFactoryHelper(
+        CatalogFactory catalogFactory, CatalogFactory.Context context) {
+      super(catalogFactory, context);
+    }
+
+    @Override
+    public void validate() {
+      validateFactoryOptions(factory, allOptions);
+      ignoreUnconsumedKeys(
+          factory.factoryIdentifier(),
+          allOptions.keySet(),
+          consumedOptionKeys,
+          deprecatedOptionKeys);
+      validateWatermarkOptions(factory.factoryIdentifier(), allOptions);
+    }
+  }
+
+  /** Validates unconsumed option keys. */
+  private static void ignoreUnconsumedKeys(
+      String factoryIdentifier,
+      Set<String> allOptionKeys,
+      Set<String> consumedOptionKeys,
+      Set<String> deprecatedOptionKeys) {
+    final Set<String> remainingOptionKeys = new HashSet<>(allOptionKeys);
+    remainingOptionKeys.removeAll(consumedOptionKeys);
+    if (!remainingOptionKeys.isEmpty()) {
+      LOG.warn(
```

Comment:
@FANNG1 Override the FactoryHelper. For the unconsumerKeys, use the log instead of the exception. So if there are some unknown keys, the Flink will not fail.

Reply:
could you add more comment to `GravitinoCatalogFactoryHelper`?

Reply:
Done. Please take a look.